lib/index.coffee

```diff
@@ -51,7 +53,38 @@ streamToString = (stream) ->
 		.on 'end', ->
 			resolve(Buffer.concat(chunks).toString())
 
+describeCounterMetricIfNotExists = (metrics, name, description, labelNames) ->
+	if metrics.exists(name) == 'false'
+		metrics.describe.counter(name, description,
+			{
+				labelNames: labelNames
+			}
+		)
+
+describeHistogramMetricIfNotExists = (metrics, name, description, labelNames, buckets) ->
+	if metrics.exists(name) == 'false'
+		metrics.describe.histogram(name, description,
+			{
+				labelNames: labelNames,
+				buckets: buckets
+			}
+		)
+
+setupMetrics = (metrics) ->
+	describeCounterMetricIfNotExists(metrics, 'gc_space_reclaimed_total', 'Disk space the GC must free', ['host'])
+	describeCounterMetricIfNotExists(metrics, 'gc_number_images_to_remove_total', 'Number of images to be removed at a given time', ['host'])
+	describeCounterMetricIfNotExists(metrics, 'gc_image_removal_errors_total',  'Number of image removal errors by type', ['host', 'statusCode'])
+	describeCounterMetricIfNotExists(metrics, 'gc_images_removed_total', 'Total number of images removed by all different methods', ['host', 'removal_type'])
+	buckets = metrics.client.exponentialBuckets(4, Math.SQRT2, 29).map(Math.round)
+	describeHistogramMetricIfNotExists(metrics, 'gc_duration_seconds', 'Milliseconds taken to free GC the reclaimed space', ['host'], buckets)
+
+recordGcRunTime = (gcMetrics, t0) ->
+	t1 = process.hrtime()
+	duration = t1[0] - t0[0] + (t1[1] - t0[1]) / 1e6
```

Comment:
```suggestion
	duration = t1[0] - t0[0] + (t1[1] - t0[1])
```

Reply:
@xginn8 `t1[0] - t0[0]` unit is `ms`, so we need to `/ 1e6` the `ns`-unit `(t1[1] - t0[1])` such that the overall result is in ms (or we could `* 1e6` `t1[0] - t0[0]`, but timing metrics in all other places are `ms` so it's good to be consistent)

Reply:
woah hold on, no, `[0]` is in seconds, so rather in fact @kaisoz this should be:

```
dt = process.hrtime(t0)
duration = dt[0]*1000 + dt[1] / 1e6
```

This parallels what we've done in `balena-api` and other places

Reply:
Gotcha, basically making sure that we return milliseconds as the metric describes :+1: